homeassistant/components/remootio/cover.py

```diff
@@ -0,0 +1,234 @@
+
+    _LOGGER.debug("config_entry [%s]" % config_entry.as_dict())
+
+    connection_options: ConnectionOptions = ConnectionOptions(
+        config_entry.data[CONF_IP_ADDRESS],
+        config_entry.data[CONF_API_SECRET_KEY],
+        config_entry.data[CONF_API_AUTH_KEY],
+    )
+    serial_number: str = config_entry.data[CONF_SERIAL_NUMBER]
+    device_class: str = config_entry.data[CONF_DEVICE_CLASS]
+
+    remootio_client: RemootioClient = await create_client(
+        hass, connection_options, _LOGGER, serial_number
+    )
+
+    async_add_entities(
+        [
+            RemootioCover(
+                serial_number, config_entry.title, device_class, remootio_client
+            )
+        ]
+    )
+
+
+class RemootioCover(cover.CoverEntity):
+    """Cover entity which represents an Remootio device controlled garage door or gate."""
+
+    __name: str
+    __unique_id: str
+    __device_class: str
+    __remootio_client: RemootioClient
+    __device_info: DeviceInfo
+
+    def __init__(
+        self,
+        unique_id: str,
+        name: str,
+        device_class: str,
+        remootio_client: RemootioClient,
+    ) -> None:
+        """Initialize this cover entity."""
+        super().__init__()
+        self.__name = name
+        self.__unique_id = unique_id
+        self.__device_class = device_class
+        self.__remootio_client = remootio_client
+        self.__device_info = DeviceInfo(
+            name="Remootio",
+            manufacturer="Assemblabs Ltd",
+            suggested_area="The first end-to-end encrypted Wi-Fi and Bluetooth enabled smart remote controller, "
+            "that lets you control and monitor your gates and garage doors using your smartphone.",
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register listeners to the used Remootio client to be notified on state changes and events."""
+        await self.__remootio_client.add_state_change_listener(
+            RemootioCoverStateChangeListener(self)
+        )
+
+        await self.__remootio_client.add_event_listener(
+            RemootioCoverEventListener(self)
+        )
+
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Terminate the used Remootio client."""
+        await self.__remootio_client.terminate()
+
+    async def async_update(self) -> None:
+        """Trigger state update of the used Remootio client."""
+        await self.__remootio_client.trigger_state_update()
+
+    @property
+    def should_poll(self) -> bool:
+        """Home Assistant shouldn't check this entity for an updated state because it notifies Home Assistant about state changes, therefore this method always returns False."""
+        return False
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the unique id of this entity. Serial number of the connected Remmotio device."""
+        return self.__unique_id
```

Comment:
Most of these properties have shorthand versions like _attr_unique_id which you should set directly from the init constructor to avoid boilerplate properties.

Reply:
Fixed.